cpp/log/cluster_state_controller-inl.h

```diff
@@ -206,6 +236,35 @@ void ClusterStateController<Logged>::CalculateServingSTH(
 }
 
 
+template <class Logged>
+void ClusterStateController<Logged>::DetermineElectionParticipation(
+    const std::unique_lock<std::mutex>& lock) {
+  CHECK(lock.owns_lock());
+  // Can't be in the election if the cluster isn't properly initialised
+  if (!actual_serving_sth_) {
+    LOG(WARNING) << "Cluster has no Serving STH - leaving election.";
+    election_->StopElection();
+    return;
+  }
+
+  // Don't want to be the master if we don't yet have the data to be able to
+  // issue new STHs
+  if (actual_serving_sth_->tree_size() >
+      local_node_state_.contiguous_tree_size()) {
+    LOG(INFO) << "Serving STH tree_size (" << actual_serving_sth_->tree_size()
+              << " < Local contiguous_tree_size ("
+              << local_node_state_.contiguous_tree_size() << ")";
+    LOG(INFO) << "Local replication too far behind to be master - "
+                 "leaving election.";
+    election_->StopElection();
+    return;
+  }
+
+  // Otherwise, make sure we're joining in the election.
+  election_->StartElection();
```

Comment:
It would have been nice to know what caused us to rejoin, but I think that might be a bit too complicated...